content/en/docs/reference/kubernetes-api/authentication-resources/token-request-v1.md

```diff
@@ -0,0 +1,165 @@
+---
+api_metadata:
+  apiVersion: "authentication.k8s.io/v1"
+  import: "k8s.io/api/authentication/v1"
+  kind: "TokenRequest"
+content_type: "api_reference"
+description: "TokenRequest requests a token for a given service account."
+title: "TokenRequest"
+weight: 2
+---
+
+`apiVersion: authentication.k8s.io/v1`
+
+`import "k8s.io/api/authentication/v1"`
+
+
+## TokenRequest {#TokenRequest}
+
+TokenRequest requests a token for a given service account.
+
+<hr>
+
+- **apiVersion**: authentication.k8s.io/v1
+
+
+- **kind**: TokenRequest
+
+
+- **metadata** (<a href="{{< ref "../common-definitions/object-meta#ObjectMeta" >}}">ObjectMeta</a>)
+
+
+- **spec** (<a href="{{< ref "../authentication-resources/token-request-v1#TokenRequestSpec" >}}">TokenRequestSpec</a>), required
+
+
+- **status** (<a href="{{< ref "../authentication-resources/token-request-v1#TokenRequestStatus" >}}">TokenRequestStatus</a>)
+
+
+
+
+
+
+## TokenRequestSpec {#TokenRequestSpec}
+
+TokenRequestSpec contains client provided parameters of a token request.
+
+<hr>
+
+- **audiences** ([]string), required
+
+  Audiences are the intendend audiences of the token. A recipient of a token must identitfy themself with an identifier in the list of audiences of the token, and otherwise should reject the token. A token issued for multiple audiences may be used to authenticate against any of the audiences listed but implies a high degree of trust between the target audiences.
+
+- **boundObjectRef** (BoundObjectReference)
+
+  BoundObjectRef is a reference to an object that the token will be bound to. The token will only be valid for as long as the bound object exists. NOTE: The API server's TokenReview endpoint will validate the BoundObjectRef, but other audiences may not. Keep ExpirationSeconds small if you want prompt revocation.
+
+  <a name="BoundObjectReference"></a>
```

Comment:
Hi @feloy. I probably missed this.
Does this HTML anchor element belong here? I see the anchor element output in a number of places but suspect the
element is ignored by the parser. I don't expect to see raw HTML anchors in the Markdown page.
Page preview:
https://deploy-preview-23294--kubernetes-io-master-staging.netlify.app/docs/reference/kubernetes-api/authentication-resources/token-request-v1/#TokenRequestSpec

Reply:
I'm using this syntax to add an anchor point into the document. I don't know any other way to add such an anchor point (see https://stackoverflow.com/questions/5319754/cross-reference-named-anchor-in-markdown)

Reply:
OK. I am trying to understand how the links are displayed in a page. Is this correct:
I could add the api-reference shortcode to another page and expect to be redirected to the start of the inline definition
of `BoundObjectReference`?
Otherwise, the link does not display in the link's source page:
https://deploy-preview-23294--kubernetes-io-master-staging.netlify.app/docs/reference/kubernetes-api/authentication-resources/token-request-v1/#TokenRequestSpec

```
{{< api-reference page="authentication-resources/token-request-v1/" anchor="BoundObjectReference" >}} 
```

As an aside, the shortcode could have a more meaningful name. What does the shortcode do? Create a single link to an API resource page, or a link to a k8s definition or resource field in an API resource page?

Reply:
I think you are mixing *Link* and *Anchor* here.

In pure HTML:
- An *Anchor* is defined with `<a name="fragment"></a>` in `pod.html`
- A *Link* is defined with `<a href="pod.html#fragment>go to the fragment on pod page</a>` in `another_page.html`

One uses the `name` attribute, the other the `href` attribute.

The anchor is not visible, it is just a mark for the browser to scroll the page to this place when you click a Link to this anchor.

In our context:
- the Anchor is defined with `<a name="BoundObjectReference"></a>` in `token-request-v1` page
- Into another page, `{{< api-reference page="authentication-resources/token-request-v1/" anchor="BoundObjectReference" >}}` is transformed into `<a href="/docs/reference/kubernetes-api/workloads-resources/pod-v1#BoundObjectReference"` which creates a link to the  `token-request-v1` page, and the browser scrolls to the part of the document where the Anchor is defined.

Reply:
Yes, right what you have added is an anchor not a link.
Though, my point is that I think a link on the page to that anchor would be useful. Related to https://github.com/kubernetes-sigs/reference-docs/issues/189

Reply:
Yes I agree, I'll try to work on this issue as soon as possible

Reply:
Hi. Just checking if there are remaining questions or concerns?
I reread some of the previous comments about URLs.
I thought a bit more about creating a data file to map the Markdown page to the location of the resource and sub-type definitions. Some types are top-level resources, some types are defined in a common directory/page, and other types are defined as sub-types within a page. The data file would provide a site map for the API object types. 
When using the shortcode, what data do authors need to locate an object:
- top-level resource (TokenRequest) page and type name (BoundObjectReference) or
- top-level resource (TokenRequest) page, sub-type name (TokenSpec), type name (BoundObjectReference)

```
kind: "TokenRequest"
page: "authentication-resources/token-request-v1"
apiVersion: "authentication.k8s.io/v1"
sub_types:
- name: TokenRequestSpec
    sub_types:
    - name: BoundObjectReference
- name: TokenRequestStatus
```